src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionResolutionEndpoint.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    internal class CodeActionResolutionEndpoint : IRazorCodeActionResolutionHandler
+    {
+        private readonly IReadOnlyDictionary<string, RazorCodeActionResolver> _resolvers;
+        private readonly ILogger _logger;
+
+        public CodeActionResolutionEndpoint(
+            IEnumerable<RazorCodeActionResolver> resolvers,
+            ILoggerFactory loggerFactory)
+        {
+            if (loggerFactory is null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _logger = loggerFactory.CreateLogger<CodeActionResolutionEndpoint>();
+
+            var resolverMap = new Dictionary<string, RazorCodeActionResolver>();
+            foreach (var resolver in resolvers ?? throw new ArgumentNullException(nameof(resolvers)))
```

Comment:
I think it may be beneficial to move the `resolvers ?? throw new ArgumentNullException(nameof(resolvers))` check up to it's own line.